scripts/subselect.py

```diff
@@ -0,0 +1,229 @@
+
+pol_grp = ap.add_argument_group("Polarization Selection")
+pol_grp.add_argument("--pols", type=str, help="Comma-separated list of polarizations to select.", default=None)
+
+ant_grp = ap.add_argument_group("Antenna Selection")
+ant_grp.add_argument("--antennas", type=str, help="Comma-separated list of antenna ranges to select.", default=None)
+ant_grp.add_argument("--calfile", type=str, help="Path to calfile to use for antenna selection.", default=None)
+
+bl_grp = ap.add_argument_group("Baseline Selection")
+bl_grp.add_argument("--bls", type=str, help="Comma-separated list of baselines to select, eg '3:5,15:27'", default=None)
+bl_grp.add_argument("--only-autos", action="store_true", help="Only select auto-correlations.")
+bl_grp.add_argument("--only-cross", action="store_true", help="Only select cross-correlations.")
+bl_grp.add_argument("--min-bl-length", type=float, help="Minimum baseline length to select, metres", default=None)
+bl_grp.add_argument("--max-bl-length", type=float, help="Maximum baseline length to select, metres", default=None)
+bl_grp.add_argument("--min-ew-length", type=float, help="Minimum EW baseline length to select, metres", default=None)
+bl_grp.add_argument("--max-ew-length", type=float, help="Maximum EW baseline length to select, metres", default=None)
+
+io_grp = ap.add_argument_group("I/O")
+io_grp.add_argument("--clobber", action="store_true", help="Overwrite output file if it exists.")
+
+check_grp = ap.add_argument_group("Checking Arguments")
+check_grp.add_argument("--check", action="store_true", help="Check output file for consistency with input file.")
+check_grp.add_argument("--check-acceptability", action="store_true", help="Check output file for consistency with input file.")
+check_grp.add_argument("--check-uvw-strict", action="store_true", help="Check whether UVW's are strictly consistent with antpos.")
+check_grp.add_argument("--check-autos", action="store_true", help="Check that autos are real")
+check_grp.add_argument("--fix-autos", action="store_true", help="Fix autos if they are not real")
+
+args = parse_args(ap)
+kw = filter_kwargs(vars(args))
+
+def select(
+    infile: str, outfile: str, 
+    freq_min=None, freq_max=None, freq_spws=None, time_min=None, time_max=None, 
+    time_idxs=None, lst_min=None, lst_max=None, lst_in_hours=False, pols=None, 
+    antennas=None, calfile=None, bls=None, only_autos=False, only_cross=False, 
+    min_bl_length=None, max_bl_length=None, min_ew_length=None, max_ew_length=None,
+    clobber=False,  check=False, check_acceptability=False, check_uvw_strict=False,
+    check_autos=False, fix_autos=False
+):
+    if not os.path.exists(infile):
+        raise FileNotFoundError(f"File {ap} does not exist.")
+    if infile != outfile and os.path.exists(outfile) and not clobber:
+        raise FileExistsError(f"File {outfile} exists and clobber is False.")
+    if infile == outfile:
+        clobber= True
```

Comment:
This feels a little dangerous. If a user chooses the same file for in and out, I get they _probably_ want it to be clobbered, but you never know. I could just imagine something terrible happening.